Add API Ref Links to readme (#798)
iotaledger/iota-sdk: README.md

## Client Usage

> The following example creates a [`Client`](https://docs.rs/iota-sdk/latest/iota_sdk/client/index.html)

Reviewer: Link to the struct ? https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.Client.html

Author: ```suggestion
The following example creates a [`Client`](https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.Client.html)
```

instance connected to
the [Shimmer Testnet](https://api.testnet.shimmer.network), and retrieves the node's information by

> calling [`Client.getInfo()`](https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.ClientInner.html#method.get_info),

Reviewer 2: ```suggestion
calling [`Client.get_info()`](https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.ClientInner.html#method.get_info),
```

> calling [`Client.get_info()`](https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.ClientInner.html#method.get_info),

Reviewer: I'd rather link to https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.Client.html#method.get_info

Author: ```suggestion
calling [`Client.get_info()`](https://docs.rs/iota-sdk/latest/iota_sdk/client/core/struct.Client.html#method.get_info),
```

## Wallet Usage

The following example will create a

> new [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/index.html) [`Account`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html)

Reviewer: Link to the struct instead

Author: ```suggestion
new [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/index.html) [`Account`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/index.html)
```

Author: ```suggestion
new [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html) [`Account`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html)
```
